examples/agent_store/api.py

```diff
@@ -82,61 +82,65 @@ async def get_agent(
                     type="memory",
                     max_chunks=5,
                     max_tokens_in_context=2048,
                     memory_bank_configs=[],
                 ),
             ]
-            agent_config = AgentConfig(
-                model=self.model,
-                instructions=textwrap.dedent(
-                    """
-                    You are an agent that can search the web (using brave_search) to answer user questions.
+            user_instructions = textwrap.dedent(
+                """
+                You are an agent that can search the web (using brave_search) to answer user questions.
 
-                    Your task is to search the web to get the information related to the provided question.
-                    Ask clarifying questions if needed to figure out appropriate search query.
-                    Cite the top sources with corresponding urls.
-                    Once you make a relevant search query, summarize the results to answer in the following format:
-                    ```
-                    This is what I found on the web:
-                    {answer}
+                Your task is to search the web to get the information related to the provided question.
+                Ask clarifying questions if needed to figure out appropriate search query.
+                Cite the top sources with corresponding urls.
+                Once you make a relevant search query, summarize the results to answer in the following format:
+                ```
+                This is what I found on the web:
+                {add answer here}
 
-                    Sources:
-                    {add sources with corresponding links}
-                    ```
-                    Do not add any other comments like greetings.
-                    """
-                ),
+                Sources:
+                {add sources with corresponding links}
+                ```
+                Do NOT add any other greetings or explanations. Just make a search call and answer in the appropriate format.
+                """
+            )
+            agent_config = AgentConfig(
+                model=self.model,
+                instructions="",
                 sampling_params=SamplingParams(
                     strategy="greedy", temperature=0.0, top_p=0.95
                 ),
                 tools=tools,
+                enable_session_persistence=True,
             )
         elif agent_type == AgentChoice.Memory:
             bank_ids = agent_params.get("bank_ids", [])
             tools = [
-                AgentConfigToolShield(
+                AgentConfigToolMemoryToolDefinition(
                     type="memory",
                     max_chunks=5,
                     max_tokens_in_context=2048,
                     memory_bank_configs=[
-                        AgentConfigToolShieldMemoryBankConfigVector(
+                        AgentConfigToolMemoryToolDefinitionMemoryBankConfigUnionMember0(
```

Comment:
lets repalce this with 
```
{"type": "vector", "bank_id": bank_id}
```
and not import this `AgentConfigToolMemoryToolDefinitionMemoryBankConfigUnionMember0`